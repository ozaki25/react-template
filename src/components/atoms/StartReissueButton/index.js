import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Container = styled.div`
  border: solid 1px #eee;
  box-shadow: 2px 2px 2px #bdbdbd;
  width: 300px;
  &:active {
    opacity: 0.5;
  }
`;

const Link = styled.a`
  text-decoration: none;
`;

const Title = styled.div`
  color: #212121;
  font-size: 14px;
  font-weight: bold;
  background-color: #eee;
  padding: 10px;
`;

const Description = styled.div`
  color: #212121;
  font-size: 12px;
  padding: 10px;
`;

function StartReissueButton({ href }) {
  return (
    <Container>
      <Link href={href}>
        <Title>利用できないキャッシュカードの再発行</Title>
        <Description>
          磁気不良、ICチップ不良、汚損、破損等で利用できなくなったキャッシュカードの再発行
        </Description>
      </Link>
    </Container>
  );
}

StartReissueButton.propTypes = {
  href: PropTypes.string,
};

StartReissueButton.defaultProps = {
  href: '#',
};

export default StartReissueButton;
