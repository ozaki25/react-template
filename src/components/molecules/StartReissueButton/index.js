import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function StartReissueButton({ onClick }) {
  return (
    <Button onClick={onClick}>
      <Title>利用できないキャッシュカードの再発行</Title>
      <Description>
        磁気不良、ICチップ不良、汚損、破損等で利用できなくなったキャッシュカードの再発行
      </Description>
    </Button>
  );
}

export const Button = styled.button`
  border: solid 1px #eee;
  box-shadow: 2px 2px 2px #bdbdbd;
  width: 300px;
  display: inline-block;
  &:active {
    opacity: 0.5;
    box-shadow: 2px 2px 2px #f5f5f5;
  }
`;

export const Title = styled.p`
  color: #212121;
  font-size: 14px;
  font-weight: bold;
  background-color: #eee;
  padding: 10px;
  text-align: left;
`;

export const Description = styled.p`
  color: #212121;
  font-size: 12px;
  padding: 10px;
  text-align: left;
`;

StartReissueButton.propTypes = {
  onClick: PropTypes.func,
};

StartReissueButton.defaultProps = {
  onClick: null,
};

export default StartReissueButton;
