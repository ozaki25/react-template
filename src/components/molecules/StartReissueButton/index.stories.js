import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import ReissueStartButton from './';

const stories = storiesOf('molecules', module);

stories.add(
  'StartReissueButton',
  withInfo('再発行の手続きをスタートするボタン')(() => (
    <ReissueStartButton onClick={action('click button')} />
  )),
);
