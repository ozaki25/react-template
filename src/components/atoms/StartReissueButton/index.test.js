import React from 'react';
import { snapshot } from '../../../../test/helpers';
import StartReissueButton from './';

snapshot('StartReissueButton', <StartReissueButton href="http://google.com/" />);
