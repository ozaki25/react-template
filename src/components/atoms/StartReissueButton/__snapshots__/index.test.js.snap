// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: StartReissueButton 1`] = `
<styled.div>
  <styled.a
    href="http://google.com/"
  >
    <styled.div>
      利用できないキャッシュカードの再発行
    </styled.div>
    <styled.div>
      磁気不良、ICチップ不良、汚損、破損等で利用できなくなったキャッシュカードの再発行
    </styled.div>
  </styled.a>
</styled.div>
`;
